### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Kovensky**
This updates the versions of dependencies that had major updates.

As updating `eslint-config-airbnb` does bring `eslint` setting changes, the code has been updated to match.

The latest `eslint-config-airbnb` doesn't yet support `eslint@3` so it was kept at 2.

### CODE REVIEW

File: src/webpack/createCompiler.js [hunk 1]
```diff
@@ -2,8 +2,9 @@ import invariant from 'invariant';
 import _ from 'lodash';
 import webpack from 'webpack';
 
-export default function createCompiler(webpackConfig, cb) {
-  invariant(arguments.length === 2, 'parameters are missing');
+const missing = () => { invariant(false, 'parameters are missing'); };
+
+export default function createCompiler(webpackConfig, cb = missing()) {
```
Comment (Kovensky) [inline src/webpack/createCompiler.js]: Nifty trick from http://www.2ality.com/2016/05/six-nifty-es6-tricks.html

File: test/configTestCases.test.js [hunk 2]
```diff
@@ -47,13 +48,13 @@ describe('ConfigTestCases', function () {
               return done(err);
             }
 
-            function _it(title, fn) {
+            function testIt(title, fn) {
```
Comment (Kovensky) [inline test/configTestCases.test.js]: `eslint-config-airbnb@^8.0.0` and up forbid dangling `_`, so I came up with slightly more descriptive names...

File: test/configTestCases.test.js [hunk 3]
```diff
@@ -73,17 +75,19 @@ describe('ConfigTestCases', function () {
...
               const bundlePath = findBundle(idx, opts);
               if (bundlePath) {
-                _require(bundlePath);
+                customRequire(bundlePath);
               }
             });
 
             // give a free pass to compilation that generated an error
             process.nextTick(done);
+            return undefined;
```
Comment (Kovensky) [inline test/configTestCases.test.js]: `consistent-return`

**Comment by zinserjan** [comment]
Wow, thank you for your effort!

Can you replace `eslint-config-airbnb` with `eslint-config-airbnb-base`? `eslint-config-airbnb/base` is deprecated and it adds some additional dependencies (`eslint-plugin-jsx-a11` & `eslint-plugin-react`) that we don't need.

**Comment by Kovensky** [comment]
@zinserjan changed to `eslint-config-airbnb-base`, which also lets `eslint` be updated to `@3`.

Fixed some new lint warnings, and also added an `.eslintrc` setting to allow requiring `peerDependencies`; otherwise it complained about mocha/webpack. Also squashed everything.

**Comment by Kovensky** [comment]
Oh, right, `eslint@3` breaks 0.12... Do you need 0.12 to stay working?

**Comment by zinserjan** [comment]
Yep, I think we should skip the lint task on node < 4.

Can you replace the following in travis.yml:
```
  - npm run lint
```

with

```
 - if [ "${TRAVIS_NODE_VERSION}" != "0.12" ]; then npm run lint; fi
```

**Pull request merged**